helix-view/src/file_watcher.rs

```diff
@@ -0,0 +1,97 @@
+use crossbeam_channel::unbounded;
+use crossbeam_channel::{select, Receiver};
+use notify::{recommended_watcher, RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::PathBuf;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver};
+
+pub struct NotifyActor {
+    sender: Sender,
+    watcher: (RecommendedWatcher, Receiver<NotifyEvent>),
+}
+
+impl NotifyActor {
+    pub fn spawn() -> NotifyHandle {
+        let (actor_tx, actor_rx) = crossbeam_channel::unbounded();
+        let (tx, rx) = unbounded_channel();
+        let actor = NotifyActor::new(Box::new(move |msg| tx.send(msg).unwrap()));
+        let thread = jod_thread::Builder::new()
+            .name("FileWatcher".to_string())
+            .spawn(move || actor.run(actor_rx))
+            .expect("Failed to spawn file watcher");
+        NotifyHandle {
+            sender: actor_tx,
+            receiver: rx,
+            thread,
+        }
+    }
+
+    fn next_event(
+        &mut self,
+        receiver: &crossbeam_channel::Receiver<ActorMessage>,
+    ) -> Option<ActorEvent> {
+        let watcher_receiver = &self.watcher.1;
+        select! {
```

Comment:
I'm not a fan of the extra dependency on crossbeam_channel, but I guess there's no way around it since we need select. Oh well..